alpine/entry.sh

```diff
@@ -1,11 +1,16 @@
 #!/bin/bash
 
+# On ResinOS 2.x devices, the hostname is set by the hostOS.
+# For backward compatibility, we only update the hostname for ResinOS 1.x devices.
 function update_hostname()
 {
-	HOSTNAME="$HOSTNAME-${RESIN_DEVICE_UUID:0:7}"
-	echo $HOSTNAME > /etc/hostname
-	echo "127.0.1.1 $HOSTNAME" >> /etc/hosts
-	hostname "$HOSTNAME"
+	if [ ${RESIN_DEVICE_UUID:0:7} != $(hostname -s | cut -c 1-7) ]; then
```

Comment:
Can this be:
```bash
if [ ${RESIN_DEVICE_UUID:0:7} != ${HOSTNAME:0:7} ]; then
```
in order to make it a bit easier to see that the same slice is being compared